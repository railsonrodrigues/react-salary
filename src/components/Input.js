import React, { Component } from 'react';

export default class Input extends Component {
  render() {
    const { id, description, className, type, step, value, readOnly } = this.props;

    return (
      <label id={ id }>
        { description }:
        <input 
          id={ id }
          className={ className } 
          type={ type }
          step={ step }
          value={ value }
          readOnly={ readOnly }
        />
      </label>
    )
  }
}
