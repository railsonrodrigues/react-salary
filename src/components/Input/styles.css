input {
  width: 200px;
  padding: 15px 30px;
  border: 1px solid #cdcdcd;
  border-radius: 5px;
}