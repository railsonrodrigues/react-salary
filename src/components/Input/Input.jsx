import React from "react";
import "./styles.css";

export default function Input({ placeholder, id }) {
  return <input id={id} type="text" placeholder={placeholder} />;
}
