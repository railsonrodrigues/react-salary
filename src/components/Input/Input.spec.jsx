import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom'
import React from 'react';
import Input from './Input';

describe('<Input />', () => {
  it('should render Input correctly', () => {
    render(<Input placeholder="Digite aqui" id="1" />)
    const input = screen.getByRole('textbox');

    expect.assertions(3);

    expect(input).toBeInTheDocument();
    expect(input).toHaveAttribute('placeholder', 'Digite aqui');
    expect(input).toHaveAttribute('id', '1');
  })
})