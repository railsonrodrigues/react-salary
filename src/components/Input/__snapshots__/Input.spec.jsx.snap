// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should match snapshot 1`] = `
<input
  class="sc-bdnxRM dBDSSv"
  id="1"
  placeholder="Digite aqui"
  type="text"
/>
`;
